### DISCUSSION TITLE
Wildcard usage in hostname filtering

### DISCUSSION THREAD

**Discussion by c2mfj** [CrowdStrike/psfalcon#429] [Q&A] [answered]
Is there a way to use a leading wildcard when filtering for hostname? I found where hostname uses partial by default but it seems to only work by matching the first characters in the string and auto appending an '*' to the end of the string. It also works if you add the trailing asterisk yourself. I am trying to perform a search like `AA*1234` or `*1234`.

Have hostnames of:
AAVWXYZ1
AAVWXYZ2
AAVWXYZ3
AAVWXYZ4
AAVWXYZ5

AALMNOP1
AALMNOP2
AALMNOP3
AALMNOP4
AALMNOP5

`Get-FalconHost -Filter "hostname:'AA'"` and  `Get-FalconHost -Filter "hostname:'AA*'"`
yields all hostnames listed

however:

`Get-FalconHost -Filter "hostname:'*VWXYZ'"` yields no results but should return the first 5 hosts in the list.   
`Get-FalconHost -Filter "hostname:*'*VWXYZ'"` returns an error of "Cannot validate argument on parameter 'Filter'"

**Comment by bk-cs** [comment]
The `hostname` filter in the Devices API defaults to partial search in the way that you used it (`hostname:'AA'` will match `AA`, `AA1`, `AA2`, etc.). If you do use an asterisk, it is only supported at the start or the end of the string. It can't be used in the middle.

**Reply by c2mfj** [reply]
That is what I was afraid of. Just checking that the following is also expected behavior:

for hostnames:
AALMNOP1234
AALMNOP

a partial search for *LMNOP only returns AALMNOP and not AALMNOP1234

**Reply by bk-cs** [reply, marked as answer]
Yes, that's correct. I recommend searching using a beginning or trailing wildcard, then filtering past that point.

```
Get-FalconHost -Filter "hostname:'AALMNOP'" -Detailed -All | Where-Object { $_.hostname -match '.+LMNOP\d+$' }
```

**Reply by bk-cs** [reply]
Side note, this type of filtering is sometimes allowed, but not with the Devices API (hence the filter validation error):
```
hostname:*'*VWXYZ'
```
